Closes #1350: Implement more flexible update hook numbering scheme. (#1351)
az-digital/az_quickstart: RELEASES.md

### End of Life
Once two more releases of Quickstart are available (approximately one year from initial release), releases will no longer be supported. This means that they will not receive any further security updates or bug fixes.
-->

## Update hook numbering convention

In order to allow update hooks to be added to different minor release branches
independently, DB update hooks implementing `hook_update_N()` should adhere to
the following update numbering convention.

**hook_update_XYZZnn()**
- X = 1 or 2 digits for Drupal core major version compatibility (e.g. `9`)
- Y = 1 digit for Quickstart major version compatibility (e.g. `2`)
- ZZ = 2 digits for Quickstart minor version compatibility (e.g. `01`)
- nn = 2 digits for sequential counting, starting with 01 (e.g. `01`)

This will allow us to continue to add DB update hooks to a minor release branch
(e.g. `2.1.x`) as needed without changing the schema version sites to be
something that would prevent this necessary  updates that are specific to a
later minor release branch (e.g. `2.2.x`) from being applied when the site is

> upgraded to the later minor release.

Reviewer: I think there are some words missing from this paragraph, not sure exactly. 🤔

Author: Updated.  Thanks!

> This will allow us to continue to add DB update hooks to a minor release branch
> (e.g. `2.1.x`) as needed without changing the schema version sites to be
> something that would prevent this necessary  updates that are specific to a
> later minor release branch (e.g. `2.2.x`) from being applied when the site is
> upgraded to the later minor release.

Author: Hopefully this sounds better.

```suggestion
This will allow us to continue to add DB update hooks to a minor release branch
(e.g. `2.1.x`) as needed without updating a site's schema version to a number
that would prevent updates included in a later release branch to be applied when
the site is upgraded.
```
